src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../../../utils/hooks/useTheme';
+import {
+  FilterButtonBar,
+  FilterButtonBarTranslation,
+} from '../FilterButtonBar';
+import { BaseFilter } from '../BaseFilter';
+import { TableFilterProps } from '../types';
+import '@itwin/itwinui-css/css/table.css';
+import DatePickerInput from './DatePickerInput';
+
+export type DateRangeTranslation = {
+  from: string;
+  to: string;
+};
+
+const defaultStrings: DateRangeTranslation = {
+  from: 'From',
+  to: 'To',
+};
+
+const defaultFormatDate = (date: Date) => {
+  const formatter = new Intl.DateTimeFormat('en-us', {
+    month: 'short',
+    day: 'numeric',
+    year: 'numeric',
+  });
+  return formatter.format(date);
+};
+
+const defaultParseInput = (text: string) => {
+  if (/^[a-z]{3}\s\d{1,2},\s\d{4}$/gi.test(text)) {
+    return new Date(text);
+  }
+  return new Date('');
+};
+
+export type DateRangeFilterProps<
+  T extends Record<string, unknown>
+> = TableFilterProps<T> & {
+  formatDate?: (date: Date) => string;
+  parseInput?: (text: string) => Date;
+  placeholder?: string;
+  translatedLabels?: DateRangeTranslation & FilterButtonBarTranslation;
+};
+
+export const DateRangeFilter = <T extends Record<string, unknown>>(
+  props: DateRangeFilterProps<T>,
+) => {
+  const {
+    column,
+    translatedLabels,
+    setFilter,
+    clearFilter,
+    formatDate = defaultFormatDate,
+    parseInput = defaultParseInput,
+    placeholder = 'MMM DD, YYYY',
```

Comment:
I'm pretty sure this should say "Mon" not "MMM". Need to research standard date format placeholders

Reply:
Oh and it would need to be localized I guess

Reply:
Changed. So user can provide his own placeholder. I don't think we can handle every language.

Reply:
Everywhere in our UX standards this short format is used, also Jon said it would be better to use it.

Reply:
ok so short format is fine but re: the original placeholder comment, I did some research and came across this Java convention: https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html

According to that, y and d should be lowercase, and also:
>  If the number of pattern letters is 3 or more, the month is interpreted as text; otherwise, it is interpreted as a number.

So I'd say go with: **MMM dd, yyyy**

Reply:
Changed.